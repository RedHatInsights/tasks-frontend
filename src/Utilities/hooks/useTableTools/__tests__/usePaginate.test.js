import { act, renderHook } from '@testing-library/react-hooks';
import usePaginate from '../usePaginate';
import items from './__fixtures__/items.fixtures';

describe('usePaginate', () => {
  it('returns a paginate configuration', () => {
    const { result } = renderHook(() => usePaginate());
    expect(result).toMatchSnapshot();
  });

  it('returns a paginate configuration', () => {
    const { result } = renderHook(() => usePaginate());

    act(() => {
      result.current.setPage(2);
    });

    expect(result.current.toolbarProps.pagination.page).toBe(2);
  });

  it('returns a paginate configuration', () => {
    const { result } = renderHook(() => usePaginate({ perPage: 5 }));

    act(() => {
      result.current.setPage(1);
    });

    const paginatedItems = result.current.paginator(items);

    expect(paginatedItems).toMatchSnapshot();
    expect(paginatedItems.length).toBe(4);
    expect(paginatedItems[1]).toBe(items[1]);
  });

  it('returns page 1', () => {
    const { result } = renderHook(() => usePaginate());

    act(() => {
      result.current.setPage(-1);
    });

    expect(result.current.toolbarProps.pagination.page).toBe(1);
  });

  it('returns empty objection with pagination disabled', () => {
    let options = { pagination: false };
    const { result } = renderHook(() => usePaginate(options));

    expect(result.current).toEqual({});
  });
});
