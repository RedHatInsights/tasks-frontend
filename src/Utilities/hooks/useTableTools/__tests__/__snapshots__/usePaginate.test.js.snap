// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`usePaginate returns a paginate configuration 1`] = `
{
  "all": [
    {
      "paginator": [Function],
      "setPage": [Function],
      "toolbarProps": {
        "pagination": {
          "onPerPageSelect": [Function],
          "onSetPage": [Function],
          "page": 1,
          "perPage": 10,
        },
      },
    },
  ],
  "current": {
    "paginator": [Function],
    "setPage": [Function],
    "toolbarProps": {
      "pagination": {
        "onPerPageSelect": [Function],
        "onSetPage": [Function],
        "page": 1,
        "perPage": 10,
      },
    },
  },
  "error": undefined,
}
`;

exports[`usePaginate returns a paginate configuration 2`] = `
[
  {
    "end_time": "2022-04-20T11:10:00",
    "id": 1,
    "initiated_by": "Michael",
    "run_date_time": "20 Apr 2022, 11:10 UTC",
    "start_time": "2022-04-20T10:10:00",
    "status": "Completed",
    "systems_count": 10,
    "task_slug": "taska",
    "task_title": "taskA",
  },
  {
    "end_time": "null",
    "id": 2,
    "initiated_by": "Michael",
    "run_date_time": "Running",
    "start_time": "2022-04-21T10:10:00",
    "status": "Running",
    "systems_count": 5,
    "task_slug": "taskb",
    "task_title": "taskB",
  },
  {
    "start_time": "2022-04
57
hehe","break!",
    "systems_count": 42,
    "task_title": "special one",
  },
  {
    "start_time": [
      42,
      "abc",
    ],
    "systems_count": 10,
    "task_title": "another one with unexpected array",
  },
]
`;
